extensions/common/vault/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/model/TokenRenewalResponsePayload.java

```diff
@@ -0,0 +1,67 @@
+/*
+ *  Copyright (c) 2023 Mercedes-Benz Tech Innovation GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Mercedes-Benz Tech Innovation GmbH - Add token rotation mechanism
+ *
+ */
+
+package org.eclipse.edc.vault.hashicorp.model;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonSetter;
+import com.fasterxml.jackson.annotation.Nulls;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+
+import java.util.List;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
```

Comment:
Why ignore unknown?

Reply:
I didn't want to serialize unnecessary properties from the API Response to 
a) reduce the performance impact
b) make the implementation less dependent on the api

If you want I can serialize all properties from the response but then I will also have to extend the tests

Reply:
There will be negligiable performance and memory impact, particularly given the context in which these operations are happening (Jackson, HTTP stack, remote call to a REST interface, non-critical code path, etc.). 

Instead of using Jackson to deserialize this to a POJO, please put it in a map and pull the values from the keys.